### ISSUE TITLE
Missing Icons on OSX?

### ISSUE THREAD

**Issue by Snipeye** [TTimo/GtkRadiant#244]
I'm on Mac 10.9, and all the bitmaps at the top (flip brush or rotate brush on x y or z, etc etc) are all missing - everything on that row.  I have all those icons on the row beneath, and I just the package contents and I can SEE the images (pngs) but they show up as the broken image picture.

-- UPDATE jdolan: The current workaround is to install gtkglext through Macports. The issue seems to be an unconventional or runtime based loading of libpng in the gdk-pixbuf stack not jiving with how we bundle shared libraries (dylibbundler). Hoping to have this properly fixed in 1.6.5.

**Comment by vikbez** [comment]
I'm using brew instead of macports, installed the packages but this does not fix the problem.
Is there any alternative ?

Thanks

**Comment by TTimo** [comment]
@jdolan are you still interested in the Mac port maintenance, or we should drop Mac from our platforms?

**Comment by donatj** [comment]
I personally think it would be foolish not to support the Mac platform as popular as it is, particularly in the United States

**Comment by LanGun** [comment]
I agree, I personally have moved on to other things but Mac culture is huge, keep it in.


Lance Gunter

**Comment by jdolan** [comment]
@TTimo Yes, absolutely. I can cut the Mac builds when 1.6.5 is ready. I'm still puttering around with Quetoo and GtkRadiant and NetRadiant ;)

**Comment by TTimo** [comment]
@jdolan I'm thinking it could be a good exercise to produce a binary of the current 1.6.5 code on OSX, so we can sanity check it.

I have 10.9 VM that I use for OSX builds with other projects that could be useful to rely on for producing builds.

**Comment by TTimo** [comment]
Is this still an issue or maybe the latest builds are no longer affected?

**Comment by jdolan** [comment]
Almost certainly still an issue. As I mentioned over IRC, I'd like to take a crack at an Xcode project for GtkRadiant. Doing so should allow me to statically link everything fairly easily, which might fix gdk-pixbuf's lameness.

**Comment by TTimo** [comment]
That's fine with me. You could statically link using scons just the same. The Gtk libraries you are working against coming from macports may be the bigger problem here.

**Comment by jdolan** [comment]
I could try adding `--static` to the few places in `config.py` where we farm out to `pkg-config` and similar programs. But honestly, I've found you can't even always trust those when trying to statically link. Xcode would almost certainly be the easier way to trial-and-error my way through that process.

Macports typically installs both static and dynamic libraries for most packages, as long as the package itself supports it. I don't *think* Macports itself is to blame here. I think `gdk-pixbuf` doesn't like, or doesn't lend itself to, `macdylibbundler`.

**Comment by ghostfreeman** [comment]
Confirming the issue as well, with dependencies I installed via Homebrew over MacPorts (that shouldn't be a factor however).

**Comment by jdolan** [comment]
@Pan7 committed some fixes that may have affected this. If you _do not_ have the previously required dependencies installed via MacPorts, could you please try this build and report if the icons issue is resolved:

http://quetoo.org/files/GtkRadiant-1.6.6.dmg

Thanks!

**Comment by TTimo** [comment]
Do we know if this was fixed?

**Comment by jdolan** [comment]
We need someone without MacPorts installed to verify @Pan7 's fix.

Here's a build cut this morning from master:

http://quetoo.org/files/GtkRadiant-1.6.6.dmg

@ghostfreeman @vikbez @LanGun Would you mind giving this build a spin?

**Comment by vikbez** [comment]
works for me with http://quetoo.org/files/GtkRadiant-1.6.6.dmg !

**Comment by jdolan** [comment]
💥

**Comment by jdolan** [comment]
Thank you again, @Pan7

**Comment by 0bsidian** [comment]
Just to confirm (for website instructions) Macports is still required but no special dependencies? Or is Macports now not required at all?

**Comment by vikbez** [comment]
Sorry but I have to unconfirm:
I tried this on another non-dev mac to be sure, but no icons...
Must not have fully cleaned deps :(

**Comment by jdolan** [comment]
Bummer.

@0bsidian we won't close this one out until it's no longer necessary to install MacPorts at all, unless you're looking to compile GtkRadiant yourself. For end-users, MacPorts will not be required.
